lib/Pipeline.js

```diff
@@ -88,8 +89,14 @@ Pipeline.processJSONWithExtras  = function(gltfWithExtras, options) {
         removeNormals(gltfWithExtras);
     }
     addDefaults(gltfWithExtras, options);
+
+    // Generate stats for unoptimized input
+    if (options.stats) {
+        gltfStatistics.printStats(gltfWithExtras, options.inputPath);
+    }
```

Comment:
As talked about offline, `gltfStatistics` should be able to work on any gltf given to it, so this function should check that `materials` or other things are `undefined` and act accordingly.